### ISSUE TITLE
In search results, maintain the faceting choices when doing a search on a new term, but not on a page refresh

### ISSUE THREAD

**Issue by ptrikutam** [greencommons/commons#196]

**Comment by gtourtellot** [comment]
@ptrikutam, to me this seems a good one to move to Up Next.

**Comment by ptrikutam** [comment]
@gtourtellot done!

**Comment by ptrikutam** [comment]
@nsanta would this require a re-architecting of the search page to accomplish? If it's going to be a larger task, we should discuss before moving forward.

**Comment by nsanta** [comment]
It needs re-architecting for accomplish the second requirement. The facets selected by the user are stored as query parameters (URL).
What I can I suggest is: Have a checkbox for reset the facets selection.

**Comment by ptrikutam** [comment]
Gotcha, let's avoid rearchitecting it, then. Can we store the search preferences in a cookie and have the search facets resettable with a checkbox like you suggested?

cc @gtourtellot @dweinberger @davidcmiller What do you think about the above solution?

**Comment by gtourtellot** [comment]
I'd be ok with that.

**Comment by dweinberger** [comment]
My preference is that the facets keep their configuration when the user
types a new phrase into the search box and does a new search (i.e., if the
person did not leave the search page and come back to it). Is that how this
would work?

That is, would the button restore the facet parameters or be a “check all”
for the facet checkboxes?

**Comment by ptrikutam** [comment]
> My preference is that the facets keep their configuration when the user types a new phrase into the search box and does a new search (i.e., if the person did not leave the search page and come back to it). Is that how this would work?

That's right. However, in the scenario we're proposing, those facets would also be maintained if the person leaves the page and comes back to it (since maintaining both levels of functionality is more complex).

@dweinberger would that be OK? Or would it be better to keep functionality as is for now?

>  That is, would the button restore the facet parameters or be a “check all” for the facet checkboxes?

It'd be a "check all" to re-set all the facets to be selected.

**Comment by dweinberger** [comment]
I personally think it’d be better (more useful, more defensible) to
maintain facet selections across page re-loads. So, yes to what you’re
proposing.

The “check all” button makes total sense.
